lib/core/diagnostics.js

```diff
@@ -111,6 +117,65 @@ if (undiciDebugLog.enabled || fetchDebuglog.enabled) {
     )
   })
 
+  // Track fetch requests
+  diagnosticsChannel.channel('undici:fetch:start').subscribe(evt => {
```

Comment:
Nice work with the `debuglog` creation 👍 
To not add too much noise, let's enable them just and only if `fetchDebuglog.enabled` is `true`, wdyt?

Reply:
Sounds good, will make the change. Thanks!

Reply:
@metcoder95 I created a new if block to check if `fetchDebuglog.enabled` is `true` however I see that the previous block has set `isClientSet = true` should I be adding this to the fetch debug log block as well?

Reply:
Yes please, otherwise will attach the listeners twice, duplicating the number of logs 👍

Reply:
@metcoder95 awesome, thank you! I have added the line